DSharpPlus/Entities/Guild/DiscordMember.cs

```diff
@@ -519,8 +519,19 @@ public async Task RevokeRoleAsync(DiscordRole role, string reason = null)
     /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
     public async Task ReplaceRolesAsync(IEnumerable<DiscordRole> roles, string reason = null)
```

Comment:
```suggestion
    /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
    /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
    /// <exception cref="ArgumentException">Thrown when attempting to add a managed role.</exception>
    public async Task ReplaceRolesAsync(IEnumerable<DiscordRole> roles, string reason = null)
```

Reply:
it should probably throw InvalidOperationException, too, cc @Plerx2493

Reply:
Agreed